### ISSUE TITLE
New-BCContainer issue 'The property 'Config' cannot be found on this object'

### ISSUE THREAD

**Issue by TheDenSter** [microsoft/navcontainerhelper#1174]
Did all the Windows updates, did update-module, and consequently wanted to recreate my container, and it crashed. 

```
PS C:\Users\Administrator\Documents\AL\TigRT> New-BCContainer `
    -accept_eula `
    -containerName $MySettings.DevContainerName `
    -artifactUrl $MySettings.ArtifactURL `
    -imageName $MySettings.ImageName `
    -licenseFile $MySettings.LicenseFile `
    -auth NavUserPassword `
    -assignPremiumPlan `
    -updateHosts `
    -accept_outdated `
    -Credential $Credential
NavContainerHelper is version 0.7.0.21
NavContainerHelper is running as administrator
Host is Microsoft Windows Server 2019 Standard - ltsc2019
Docker Client Version is 19.03.5
Docker Server Version is 19.03.5
Removing container MySandBox
Removing MySandBox from host hosts file
Removing C:\ProgramData\NavContainerHelper\Extensions\MySandBox
ArtifactUrl and ImageName specified
Fetching all docker images
Downloading application artifact /sandbox/16.3.14085.15087/us
Downloading C:\Users\Administrator\AppData\Local\Temp\4\5f46a313-0e7b-4b5d-b269-4715ef4fcb8a.zip
Unpacking application artifact to tmp folder using Expand-Archive
Image sandbox:sandbox-16.3.14085.15087-us doesn't exist
Building image sandbox:sandbox-16.3.14085.15087-us based on https://bcartifacts.azureedge.net/sandbox/16.3.14085.15087/us
Pulling latest image mcr.microsoft.com/dynamicsnav:10.0.17763.1339-generic
10.0.17763.1339-generic: Pulling from dynamicsnav
4612f6d0b889: Already exists
fc1b9e59edad: Already exists
8a943217e0a7: Already exists
6dc988e2efd9: Already exists
4ce1ae7b9cfc: Already exists
32fad1b53b36: Already exists
4520c21c9ce1: Already exists
daed7e87ad55: Already exists
4e2b24c036f3: Already exists
90125d17086d: Already exists
fff617907868: Already exists
fec714f503c8: Already exists
85f5eb990de2: Already exists
a5dc8e2a8759: Already exists
a742cf654a58: Already exists
e5e9144b0b80: Already exists
fb9edca993e6: Already exists
e47596dccb47: Pulling fs layer
1bc1f7ed5ea6: Pulling fs layer
e6504ed8859c: Pulling fs layer
e6504ed8859c: Download complete
e47596dccb47: Download complete
e47596dccb47: Pull complete
1bc1f7ed5ea6: Verifying Checksum
1bc1f7ed5ea6: Download complete
1bc1f7ed5ea6: Pull complete
e6504ed8859c: Pull complete
Digest: sha256:42ac153250c485866b0872ee60ac57a73ee7ea4058fbe8c13ced2d5be1e20779
Status: Downloaded newer image for mcr.microsoft.com/dynamicsnav:10.0.17763.1339-generic
Generic Tag: 0.1.0.12
Container OS Version: 10.0.17763.1339 (ltsc2019)
Host OS Version: 10.0.17763.1369 (ltsc2019)
Downloading platform artifact /sandbox/16.3.14085.15087/platform
Downloading C:\Users\Administrator\AppData\Local\Temp\4\a8f60c53-e9b2-4ac8-ab9b-d3760dc85ddd.zip
Unpacking platform artifact to tmp folder using Expand-Archive
Downloading Prerequisite Components
Downloading c:\bcartifacts.cache\sandbox\16.3.14085.15087\platform\Prerequisite Components\Open XML SDK 2.5 for Microsoft Office\OpenXMLSDKv25.msi
Downloading c:\bcartifacts.cache\sandbox\16.3.14085.15087\platform\Prerequisite Components\IIS URL Rewrite Module\rewrite_2.0_rtw_x64.msi
Downloading c:\bcartifacts.cache\sandbox\16.3.14085.15087\platform\Prerequisite Components\DotNetCore\DotNetCore.1.0.4_1.1.1-WindowsHosting.exe
Copying Platform Artifacts
Copying Database
Copying Licensefile
Copying ConfigurationPackages
Copying Extensions
Copying Applications.US
c:\bcartifacts.cache\tmp637316262654037275
Sending build context to Docker daemon  1.755GB
Step 1/6 : FROM mcr.microsoft.com/dynamicsnav:10.0.17763.1339-generic
 ---> 96fd9fe8dd2c
Step 2/6 : ENV DatabaseServer=localhost DatabaseInstance=SQLEXPRESS DatabaseName=CRONUS IsBcSandbox=Y artifactUrl=https://bcartifacts.azureedge.net/sandbox/16.3.14085.15087/us
 ---> Running in d20b3917dc1a
Removing intermediate container d20b3917dc1a
 ---> 991cd4ce79a3
Step 3/6 : COPY my /run/
 ---> 3db4acfd38ef
Step 4/6 : COPY NAVDVD /NAVDVD/
 ---> 70d36de3689e
Step 5/6 : RUN \Run\start.ps1 -installOnly
 ---> Running in 0758da8b6c0a
hcsshim::CreateComputeSystem 0758da8b6c0a05e44e0c2bfc29056d2212fd06c7a564f8dc2065a07c414b1c3d: The request is not supported.
(extra info: {"SystemType":"Container","Name":"0758da8b6c0a05e44e0c2bfc29056d2212fd06c7a564f8dc2065a07c414b1c3d","Owner":"docker","IgnoreFlushesDuringBoot":true,"LayerFolderPath":"C:\\ProgramData\\docker\\windowsfilter\\0758da8b6c0a05e44e0c2bfc29056d2212fd06c7a564f8dc2065a07c414b1c3d","Layers":[{"ID":"866b4a4c-0d78-5098-92f9-c071fb889b29","Path":"C:\\ProgramData\\docker\\windowsfilter\\0c6780454b13ae8ade59f8bd4744fd527e1d6a4e104d3546cdcaf57ed1c5a806"},{"ID":"d3b6fe60-9b3c-5db5-bd22-d7f066d7f90c","Path":"C:\\ProgramData\\docker\\windowsfilter\\a1c6b6e91d64e864e53b7d597a287d98a4b09a8858e693147f8727c38c950de2"},{"ID":"a16f5358-d832-5e9d-ab02-519d57b0167d","Path":"C:\\ProgramData\\docker\\windowsfilter\\632dd3a54e40044bdd6db2e83c970432f90868ab116a4a43d9e7db35b8e7d0d3"},{"ID":"6280aaff-b81d-5dc5-8b5f-15227e051825","Path":"C:\\ProgramData\\docker\\windowsfilter\\4a48094a9a5b98029fc94a0bb09303bfdced8706937885384638b6e7818ade2d"},{"ID":"5b82a2d9-f621-5af6-b61a-443bab04cab2","Path":"C:\\ProgramData\\docker\\windowsfilter\\b01c0b37b0845a5df510da77fdf19e600366b2ee73e0e861e263d55020e5f554"},{"ID":"b35350c9-49f7-5380-82c7-94f18d32a9d4","Path":"C:\\ProgramData\\docker\\windowsfilter\\9187a6561b7a9bca0768681c0f780f1dd6f1380535f949618b136134b248409a"},{"ID":"7e46d176-4645-57b2-bc33-abff33dbbbbd","Path":"C:\\ProgramData\\docker\\windowsfilter\\0bdd13f7e5d742dcdeef7dc49d43529204cbcefe58d700a8704c0ede3ca8f158"},{"ID":"85cd2d67-cb33-5c62-a388-cca7d46ac944","Path":"C:\\ProgramData\\docker\\windowsfilter\\a7645152f2564d348afb075657f111ac6f2949411d924a9a07b95000564db969"},{"ID":"d25aa6ca-25ea-5d42-91fa-2e83cfa18832","Path":"C:\\ProgramData\\docker\\windowsfilter\\c7ab396665b0046a7bc12e2325a347f45b0d492e8102aba18b0b899e84cfb957"},{"ID":"b1c15ec1-9076-53da-8a1f-a6ef43331801","Path":"C:\\ProgramData\\docker\\windowsfilter\\4aea3a4c0d1a042888b76f9104be8f2aad41f689fc5eabce20457d3efcc78ce3"},{"ID":"80ae3c9f-6e71-5647-b065-03d9a5d0aec7","Path":"C:\\ProgramData\\docker\\windowsfilter\\8148be151593ad169c3d0987478b753cc3a297a680e4b482b8b78a22e3576cb7"},{"ID":"00e619e3-4c9c-550f-8942-3debcffc9f3c","Path":"C:\\ProgramData\\docker\\windowsfilter\\b5396c02c10fd833147a42757dd35f82a97b60939775e7745de0928c7f530af2"},{"ID":"69ad0190-4113-58ca-ba97-a5e2356653c3","Path":"C:\\ProgramData\\docker\\windowsfilter\\d84c2ad3d2048ff549c19b43bbbe727d6415ff4ca1e6195fbd33825b6ea16774"},{"ID":"3e6e2cc3-73d2-5cc5-a98f-e96ce11da165","Path":"C:\\ProgramData\\docker\\windowsfilter\\ea0c09f75fd78e9aed1bdf89a6a1c644e590e70b55eda92bc9d2a8b07face4e4"},{"ID":"829bd361-9a74-514e-a191-e3a9110e5e70","Path":"C:\\ProgramData\\docker\\windowsfilter\\0d87e4ebaa8da13ebc191d5781661b6207267aff11a78dd94d83c6b5fe63e9cc"},{"ID":"681f6b3a-7443-5daf-8cc9-5c0fd2e207fd","Path":"C:\\ProgramData\\docker\\windowsfilter\\48e8bbd092da1c7203bbf9845ff81f372559c0291569c5d4b958456918f5afb7"},{"ID":"d1b56224-b212-552e-8e78-cbdf66f269e7","Path":"C:\\ProgramData\\docker\\windowsfilter\\5c7785b35778870b54bba21bb55c26aafd485bf5c067d9f657bb09ae5ed3e150"},{"ID":"e6751034-5a72-5446-bc94-37468c18ca52","Path":"C:\\ProgramData\\docker\\windowsfilter\\4384aa8db9e54dbd1f02aafa50acbc19ca76ffc5c21787eaa4685d18e06d7c05"},{"ID":"d15d42a4-23ee-5420-88ed-6c7c94ee28bb","Path":"C:\\ProgramData\\docker\\windowsfilter\\2e42b7cb0cab6b1cb4cfe438bb1760c5cbf480a9e7fc3b67a12f56aaf18fc57a"},{"ID":"461e0c75-e9e9-5583-a49d-01f0c24b94e5","Path":"C:\\ProgramData\\docker\\windowsfilter\\e1166247294a0bfac4cd0713de8c4fde49091fb007560241df4efa975cf1e1e8"},{"ID":"c36645c5-8417-52ed-a6bf-4fdd46ac1a94","Path":"C:\\ProgramData\\docker\\windowsfilter\\477a3a7681a0ecf54ce670be749bda5c19b46ae19aa955742f76824fd86bee17"},{"ID":"0635d713-9725-50d8-90c9-436133a9f271","Path":"C:\\ProgramData\\docker\\windowsfilter\\e6817b17b34882b82d37bc10bff2a9b188610b9112d731ceb940c9717604d4a3"},{"ID":"bd70612c-4ce0-5966-b04d-c4fa1838f738","Path":"C:\\ProgramData\\docker\\windowsfilter\\0935d7b6e4e6f1e17d413227953f28048c20e9ba71ff72d9add9b95df61d9de6"}],"MemoryMaximumInMB":4096,"HostName":"0758da8b6c0a","HvPartition":true,"EndpointList":["ABC1CAFC-A1D9-40EF-8DE5-20B62AC0BCB1"],"HvRuntime":{"ImagePath":"C:\\ProgramData\\docker\\windowsfilter\\e6817b17b34882b82d37bc10bff2a9b188610b9112d731ceb940c9717604d4a3\\UtilityVM"},"AllowUnqualifiedDNSQuery":true})
Building image took 647 seconds
Using image sandbox:sandbox-16.3.14085.15087-us
Error: No such object: sandbox:sandbox-16.3.14085.15087-us
New-BCContainer : The property 'Config' cannot be found on this object. Verify that the property exists.
At line:1 char:1
+ New-BCContainer `
+ ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [New-NavContainer], RuntimeException
    + FullyQualifiedErrorId : PropertyNotFoundStrict,New-NavContainer
```

Ran the same script again, same result. Removed all the images and ran it yet again, with the same result.

**Comment by freddydk** [comment]
These docker errors are very hard to troubleshoot, when I run the same script here, it obviously works.
You could try 2 things:
1. Remove the -imagename parameter and run directly instead of building the image first
2. use -isolation hyperv

**Comment by TheDenSter** [comment]
```
Container OS Version: 10.0.17763.1339 (ltsc2019)
Host OS Version: 10.0.17763.1369 (ltsc2019)
```
It looks like there is no generic image for 1369, so since I was scolded for not installing all Windows updates the last time I had a problem, I thought I'd uninstall the latest Windows update. I once again removed all images and containers. It is downloading the image again. If that doesn't work I will try your suggestion.

**Comment by freddydk** [comment]
Ahh - I didn't notice that there was a difference in the release numbers.
It does look very similar to https://github.com/microsoft/navcontainerhelper/issues/1135
which also was a newer host.
It looks like the docker build process doesn't work using process isolation if the versions doesn't match 100%
My assumption is. that the two workarounds above would work.

**Comment by TheDenSter** [comment]
Uninstalling windows update 1369 worked. I guess we need some sort of indication when there is a new generic image before installing these Windows updates. The issue is not solved, but the workaround works for me. I don't have time to track this so I'll close it

**Comment by freddydk** [comment]
This command:
```
(get-navcontainerimagetags -imageName "mcr.microsoft.com/windows/servercore").tags | Where-Object { $_ -like '10.0.17763.????' }
```
returns a list of all Windows ServerCore Container OS versions shipped by the Windows Team.

```
10.0.17763.1039
10.0.17763.1040
10.0.17763.1098
10.0.17763.1158
10.0.17763.1217
10.0.17763.1282
10.0.17763.1339
```

These are the versions we have shipped as generic images as well - I don't know where 1369 comes from.
Is that an insider build?

**Comment by TheDenSter** [comment]
I just ran the regular Windows update this morning (manually, like I've been doing multiple times a week for the past 20 years) and that's what it installed. I don't think I have any insider build set up anywhere.

**Comment by TheDenSter** [comment]
Seems like it automatically installed it again, even though I uninstalled it manually earlier today.
![image](https://user-images.githubusercontent.com/23434319/88858582-0ca66400-d1ad-11ea-9e24-01f3b86e67aa.png)
This $#!+ is starting to seriously tick me off

**Comment by freddydk** [comment]
I have created a Windows Server and is applying this update now.
Assumption is that the two workarounds above should work.

**Comment by TheDenSter** [comment]
And it is downloading/installing another one
![image](https://user-images.githubusercontent.com/23434319/88858867-8a6a6f80-d1ad-11ea-8fba-03396558ba7e.png)

**Comment by TheDenSter** [comment]
So it looks like I have nothing to say about my computer anymore. Windows automatically re-applied the update
![image](https://user-images.githubusercontent.com/23434319/88859662-03b69200-d1af-11ea-96c7-bddecb391e3f.png)
I'll keep the containers alive until they don't work anymore, I have a deadline to meet

**Comment by freddydk** [comment]
This is caused by host and container OS' being different and Hyper-V not being installed.
During image build - the default becomes hyperv isolation because the OS' differ.
The 2 workarounds that should work is not the 2 workarounds above, but instead:
1. -isolation process
2. remove -imagename parameter

I will make a fix in the containerhelper, where I only default to hyperv if hyperv is installed.

**Comment by TheDenSter** [comment]
How come this has never caused an issue before? Could I not just enable hyper-v?

**Comment by freddydk** [comment]
You would need nested virtualization to run hyperv inside hyperv.

This issue only appears if:
- You are running Windows Server 2019 without hyper-v
- You are using -artifactUrl AND -imageName (this has only been around for a month)
- Your host has a newer version than what is available as a container

So not very common IMO - but might happen more over the next days when people get 1369 and Windows Team hasn't released a server core for 1369, which is why I will ship a fix today.

adding -isolation process solves the issue.

**Comment by TheDenSter** [comment]
Alright thanks Freddy. I have some work I need to finish so I can't risk not having a working container. I'll try the fix once I'm done with that, probably over the weekend

**Comment by freddydk** [comment]
Shipped in 0.7.0.22
